examples/instances.rs

```diff
@@ -83,11 +85,23 @@ impl<const N: usize, const A: usize, F: PrimeField> StepCircuit<A, F> for Instan
     /// Return `z_out` result
     fn synthesize_step(
         &self,
-        _config: Self::Config,
-        _layouter: &mut impl Layouter<F>,
+        config: Self::Config,
+        layouter: &mut impl Layouter<F>,
         z_i: &[AssignedCell<F, F>; A],
     ) -> Result<[AssignedCell<F, F>; A], SynthesisError> {
-        Ok(z_i.clone())
+        for (input, instance) in z_i.iter().zip(config.instances) {
+            layouter.constrain_instance(input.cell(), instance, 0)?;
+        }
+
+        let mut z_o = z_i.clone();
+
+        // If `FAIL` is set, we will reverse the input and thus the check will break already at step 1
+        // with public input
+        if FAIL {
```

Comment:
what is the purpose of FAIL?

Reply:
We should not only test the success case, but also the case where the ivc must fail. 

Here FAIL is flag for reverse `z_i`, but not for reverse public input. And we have copy constraint between them.

If `FAIL=true` for half of the steps - step-circuit will be successful, but for half not. And we must get an error at the end of the example run.